codespan-reporting/src/term/views.rs

```diff
@@ -135,6 +135,34 @@ where
                 }
             };
 
+            let lines_before = usize::min(start_line_index, self.config.before_label_lines);
+            for offset in (1..lines_before + 1).rev() {
```

Comment:
Why this `min` call; shouldn't there be no context shown if `before_label_lines` is zero? The way this is implemented it would add lines 1 to the line this label starts on.

`rev`'ing the range/iterator does not change the outcome because `get_or_insert_line` will always insert the line respective of the line index. I think this could be removed.

Reply:
The idea of the `min` here is to handle the edge case of being at the border of the file:
assume before_label_lines is `2`, but the label starts at the second line of the file (`start_line_index = 1`).
Now, we can only print 1 line of context before the label. The min call means that if there are less lines before the label than should be printed, we just print the amount of lines that _should_ get printed.
i.e. in the case of start_line_index = 200, the min value would be `2` - thus 2 lines of context get shown. 

This definitely needs a comment now that I think about it ^^'

Assuming that lines_before results in `0` (because the configuration sets it to 0, or we're on the first line) the loop would iterate over 1..1 - which yields no values, and thus do nothing. If I'm not missing something, this should be the desired behaviour.

`rev`ing the iterator has the purpose of making sure that if anything goes wrong with the line_range, we don't do unnecessary iterations (if line n-1 doesn't work, then line n-2 also won't work). Assuming that a failure of `line_range` is sufficiently unlikely, this could of course be removed.

Reply:
I made some changes to this: I think this is a perfect case for using `checked_sub`. IMHO this is easier to understand than `min`.
Also in that case using `rev` actually does the opposite of what you wanted so removed that too.